Index: AsmJit/AsmJit/Core/Func.h
===================================================================
--- AsmJit/AsmJit/Core/Func.h	(revision 419)
+++ AsmJit/AsmJit/Core/Func.h	(working copy)
@@ -259,7 +259,7 @@
 
   template<typename T>
   inline void setReturnTypeT()
-  { setReturnValueRaw(TypeId<ASMJIT_TYPE_TO_TYPE(T)>::Id); }
+  { setReturnTypeRaw(TypeId<ASMJIT_TYPE_TO_TYPE(T)>::Id); }
 
   template<typename T>
   inline void setArgumentT(uint32_t id)
