Index: AsmJit/AsmJit/Core/StringBuilder.h
===================================================================
--- AsmJit/AsmJit/Core/StringBuilder.h	(revision 419)
+++ AsmJit/AsmJit/Core/StringBuilder.h	(working copy)
@@ -38,7 +38,7 @@
   ASMJIT_API StringBuilder() ASMJIT_NOTHROW;
   ASMJIT_API ~StringBuilder() ASMJIT_NOTHROW;
 
-  inline StringBuilder(_DontInitialize&) ASMJIT_NOTHROW {}
+  inline StringBuilder(_DontInitialize const&) ASMJIT_NOTHROW {}
 
   // --------------------------------------------------------------------------
   // [Accessors]
